diff --git a/GeosCore/sulfate_mod.F90 b/GeosCore/sulfate_mod.F90
index 84794af..9bd3f12 100644
--- a/GeosCore/sulfate_mod.F90
+++ b/GeosCore/sulfate_mod.F90
@@ -2715,22 +2715,29 @@ CONTAINS
 
        ! Isolate H2SO4 for reaction with dust    tdf 3/6/2K9
        IF ( LDSTUP ) THEN
+          ! TMS 2020/12/8 - Temporary - manually set the index of PSO4
+          id_PSO4 = 5
           ! Compute gas phase SO4 production again, as in offline case
           ! RK1: SO2 + OH(g) [s-1]  (rjp, bmy, 3/23/03)
           M    = State_Met%AIRDEN(I,J,L) * F
-          KK   = K0 * M / Ki
-          F1   = ( 1.e+0_fp + ( LOG10( KK ) )**2 )**( -1 )
-          RK1  = ( K0 * M / ( 1.e+0_fp + KK ) ) * 0.6e+0_fp**F1 * &
-                   GET_OH( I, J, L, Input_Opt, State_Chm, State_Met)
-          RKT  =  RK1 * DTCHEM  ! [unitless] (bmy, 6/1/00)
-          SO20 = SO2_cd
-          H2SO4_cd = SO20 * ( 1.e+0_fp - EXP( -RKT ) )
+!          KK   = K0 * M / Ki
+!          F1   = ( 1.e+0_fp + ( LOG10( KK ) )**2 )**( -1 )
+!          RK1  = ( K0 * M / ( 1.e+0_fp + KK ) ) * 0.6e+0_fp**F1 * &
+!                   GET_OH( I, J, L, Input_Opt, State_Chm, State_Met)
+!          RKT  =  RK1 * DTCHEM  ! [unitless] (bmy, 6/1/00)
+!          SO20 = SO2_cd
+!          H2SO4_cd = SO20 * ( 1.e+0_fp - EXP( -RKT ) )
+
+          ! Convert State_Diag%Prod from [molec/cm3/s] to [v/v/timestep].
+          KPPProd0 = State_Diag%Prod(I, J, L, id_PSO4)
+          H2SO4_cd = KPPProd0 / M * DTCHEM
 
           !tdf Reset these constants to zero to avoid any problems below
           M   = 0.e+0_fp
-          KK  = 0.e+0_fp
-          F1  = 0.e+0_fp
-          RK1 = 0.e+0_fp
+          KPPProd0 = 0.e+0_fp
+!          KK  = 0.e+0_fp
+!          F1  = 0.e+0_fp
+!          RK1 = 0.e+0_fp
        ENDIF
 
        !==============================================================
